### PR TITLE
feat: only collecting all tables on demand in influxql planner

### PR DESCRIPTION
**Pull request by Rachelint**
## Which issue does this PR close?

Closes #
Part of #830 

## Rationale for this change
Now, we collect and keep all tables in influxql planner even though it is unused.
It is wasteful to do this. So in this pr, I make it just do such works on demand.
 

## What changes are included in this PR?
+ Add `all_tables` method in `MetaProvider`.
+ Just call `MetaProvider::all_tables` to get tables while they are actually needed(such as while influxql's `SchemaProvider::table_names` is called).

## Are there any user-facing changes?
None.

## How does this change test
Test by ut.

### CODE REVIEW

File: query_frontend/src/influxql/planner.rs [hunk 1]
```diff
@@ -58,21 +66,53 @@ impl<'a, P: MetaProvider> SchemaProvider for InfluxQLSchemaProvider<'a, P> {
     }
 
     fn table_names(&self) -> Vec<&'_ str> {
-        self.tables.keys().map(|k| k.as_str()).collect::<Vec<_>>()
+        let tables = match self
```
Comment (jiacai2050) [inline query_frontend/src/influxql/planner.rs:69]: You can also to implement `table_exists` method, then it won't call `table_names`.

There is a `table` method in `MetaProvider` for this purpose.
```
fn table(&self, name: TableReference) -> Result<Option<TableRef>>;
```

File: query_frontend/src/provider.rs [hunk 2]
```diff
@@ -72,6 +84,9 @@ pub trait MetaProvider {
 
     /// Get udaf by name.
     fn aggregate_udf(&self, name: &str) -> Result<Option<AggregateUdf>>;
+
+    /// Return all tables.
+    fn all_tables(&self) -> Result<Vec<TableRef>>;
```
Comment (jiacai2050) [inline query_frontend/src/provider.rs:89]: Add comments when this is used now, and warn users this maybe incur expensive cost.

File: query_frontend/src/provider.rs [hunk 3]
```diff
@@ -138,6 +153,34 @@ impl<'a> MetaProvider for CatalogMetaProvider<'a> {
...
+    // refactor the tables collecting procedure.
+    fn all_tables(&self) -> Result<Vec<TableRef>> {
+        let catalog = match self
+            .manager
+            .catalog_by_name(self.default_catalog)
+            .with_context(|| FindCatalog {
+                name: self.default_catalog,
+            })? {
+            Some(catalog) => catalog,
+            None => return Ok(Vec::default()),
```
Comment (jiacai2050) [inline query_frontend/src/provider.rs:167]: Why not throw error when no catalog isn't found?

File: query_frontend/src/influxql/planner.rs (hunk 1 above)
Comment (Rachelint) [reply query_frontend/src/influxql/planner.rs:69]: Thanks for reminding, I forgot this default impl method...

File: query_frontend/src/provider.rs (hunk 3 above)
Comment (Rachelint) [reply query_frontend/src/provider.rs:167]: I think it again, better to return error really.

**Pull request merged**